GPU/Software/SoftGpu.cpp

```diff
@@ -951,14 +953,20 @@ bool SoftGPU::GetCurrentFramebuffer(GPUDebugBuffer &buffer, GPUDebugFramebufferT
 
 	buffer.Allocate(x2 - x1, y2 - y1, fmt);
 
-	const int depth = fmt == GE_FORMAT_8888 ? 4 : 2;
-	const u8 *src = fb.data + stride * depth * y1;
-	u8 *dst = buffer.GetData();
-	const int byteWidth = (x2 - x1) * depth;
-	for (int y = y1; y < y2; ++y) {
-		memcpy(dst, src + x1, byteWidth);
-		dst += byteWidth;
-		src += stride * depth;
+	if(fmt == GE_FORMAT_8888) {
+		u32 *dst = (u32 *)buffer.GetData();
+		for (int y = y1; y < y2; ++y) {
+			for (int x = x1; x < x2; ++x) {
+				*dst++ = fb.Get32(x, y, stride);
```

Comment:
We may want to check that Linux/Windows still optimize this to the same code as before for here and above.  I know the software renderer is slow anyway, but best not to pile onto it...

Also nit: `if (fmt == GE_FORMAT_8888) {`.

-[Unknown]

Reply:
yeah we can't always assume the compiler will optimize this back into a memcpy, although most of them would.
`GetCurrentFramebuffer` is only used for the debugger, and for taking savestate screenshots, not during rendering.

Reply:
haha yeah it is quite a challenging situation. it is fine as a dev, their engineers on the other hand must have had a hard time designing this thing heh.
they do offer some tools at least to deal with it, like DMA's with endian swapping, and color channels swizzeling, load/store instructions with swapping, etc.
too bad gcc lacks an `__attribute__((little_endian))` extension xD (there is `scalar_storage_order` for C at least)

it is beneficial in this case though since there is no need to swap PSP generated image data during display.